webapps/api/backend/views/vdisks.py

```diff
@@ -263,23 +279,12 @@ def create_from_template(self, vdisk, name, storagerouter_guid):
     @load(VDisk)
     def get_target_storagerouters(self, vdisk, hints):
         """
-        Gets all possible target Storage Routers for a given vDisk (e.g. when cloning or creating from template)
+        Gets all possible target Storage Routers for a given vDisk (e.g. when cloning, creating from template or moving)
         """
-        storagerouter_guids = None
-        storagerouters = {}
-        if vdisk.vpool is not None:
-            vpool = vdisk.vpool
-            this_storagerouter_guids = set()
-            for storagedriver in vpool.storagedrivers:
-                this_storagerouter_guids.add(storagedriver.storagerouter_guid)
-                if hints['full'] is True:
-                    storagerouters[storagedriver.storagerouter_guid] = storagedriver.storagerouter
-            if storagerouter_guids is None:
-                storagerouter_guids = list(this_storagerouter_guids)
-            else:
-                storagerouter_guids = list(this_storagerouter_guids & set(storagerouter_guids))
-            return storagerouter_guids if hints['full'] is False else [storagerouters[guid] for guid in storagerouter_guids]
-        return []
+        storagerouters = [] if vdisk.vpool is None else [sd.storagerouter for sd in vdisk.vpool.storagedrivers]
+        if hints.get('full', False) is True:
```

Comment:
Just return the storagerouters immediately, I don't see benefit in this approach